src/EventStore.php

```diff
@@ -359,7 +350,7 @@ public function rollback(): void
 
         $this->actionEventEmitter->dispatch($event);
 
-        $this->recordedEvents = new ArrayIterator();
+        $this->resetRecordedEvents();
```

Comment:
Of course, my reason was to be consistent what "recordedEvents" should be. But i agree with the performance reasons.